packages/fxa-settings/src/models/pages/index/query-params.test.ts

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { queryStringToGenericData } from '../../../lib/model-data';
+import { IndexQueryParams } from './query-params';
+
+/**
+ * These tests spot check certain validation rules. We don't necessary want to revalidate that
+ * the class-validator works, but there are certain params that we might want to keep an eye on
+ * due to curve balls thrown in the past.
+ */
+
+describe('SigninQueryParams checks', function () {
+  function validate(query: string) {
+    return new IndexQueryParams(queryStringToGenericData(query)).tryValidate();
+  }
+
+  it('checks email', () => {
+    // TBD? Should we support non uri encoded components?
```

Comment:
Unencoded + sign in email query params does not error in backbone, so if we don't support at least that character we might introduce errors that would block some users from logging in